### PR TITLE
[acl] Add tests for matching ICMP type/code

### LINKED ISSUE #3635: Need new test: ACL on icmp
**Description**

Test ACL rules that matching:
- [ ] ICMP type/code
- [ ] ICMPv6 type/code

### PR DESCRIPTION
**Pull request by daall**
Signed-off-by: Danny Allen <daall@microsoft.com>

### Description of PR

Summary:
Fixes #3635 

### Type of change

- [ ] Bug fix
- [ ] Testbed and Framework(new/improvement)
- [x] Test case(new/improvement)

#### What is the motivation for this PR?
ICMP type/code match was recently added to acl-loader, so we need to validate the functionality.

#### How did you do it?
Added a new test case and new rules for IPv4 and IPv6.

#### How did you verify/test it?
Ran new test cases locally.

### CODE REVIEW

File: tests/acl/Untitled-1 [hunk 1]
```diff
@@ -0,0 +1,8 @@
+ACTIVE:
```
Comment (theasianpianist) [inline tests/acl/Untitled-1:1]: Does this file belong?

Comment (daall) [reply tests/acl/Untitled-1:1]: lol no

File: tests/acl/test_acl.py [hunk 2]
```diff
@@ -612,14 +612,6 @@ def test_ingress_unmatched_blocked(self, setup, direction, ptfadapter, ip_versio
         pkt = self.tcp_packet(setup, direction, ptfadapter, ip_version)
         self._verify_acl_traffic(setup, direction, ptfadapter, pkt, True, ip_version)
 
-    def test_egress_unmatched_forwarded(self, setup, direction, ptfadapter, ip_version, stage):
```
Comment (wangxin) [inline tests/acl/test_acl.py:615]: Why removed this case?

Comment (bingwang-ms) [reply tests/acl/test_acl.py:615]: I don't think this case should be removed. I'd like to keep it.

**Comment by qiluo-msft** [comment]
>     def test_ingress_unmatched_blocked(self, setup, direction, ptfadapter, ip_version, stage):

function name conflicting with the below one?

---
Refers to: tests/acl/test_acl.py:607 in 47c3017. [](commit_id = 47c3017fd8c968c9abdaf03ecfa496fc8e8a9e5a, deletion_comment = False)

**Comment by qiluo-msft** [comment]
>     def test_ingress_unmatched_blocked(self, setup, direction, ptfadapter, ip_version, stage):

The 2 functions are exactly the same.

---
In reply to: [856450422](https://github.com/Azure/sonic-mgmt/pull/3485#issuecomment-856450422) [](http://example.com/codeflow?ancestors=856450422)

---
Refers to: tests/acl/test_acl.py:607 in 47c3017. [](commit_id = 47c3017fd8c968c9abdaf03ecfa496fc8e8a9e5a, deletion_comment = False)

**Review by qiluo-msft — CHANGES_REQUESTED** [review]
As comments

**Comment by bingwang-ms** [comment]
> > ```
> > def test_ingress_unmatched_blocked(self, setup, direction, ptfadapter, ip_version, stage):
> > ```
> 
> function name conflicting with the below one?
> 
> Refers to: tests/acl/test_acl.py:607 in [47c3017](https://github.com/Azure/sonic-mgmt/commit/47c3017fd8c968c9abdaf03ecfa496fc8e8a9e5a). [](commit_id = [47c3017](https://github.com/Azure/sonic-mgmt/commit/47c3017fd8c968c9abdaf03ecfa496fc8e8a9e5a), deletion_comment = False)

Yes, my mistake when editing manually. Fixed.

**Comment by qiluo-msft** [comment]
Does this PR resolve https://github.com/Azure/sonic-mgmt/issues/3635?

**Comment by bingwang-ms** [comment]
> Does this PR resolve #3635?

Yes. Both ICMP v4 and v6 tests are covered.

**Review by wangxin — APPROVED** [review]

**Pull request merged**